### PR TITLE
azure ipi private workflow

### PR DESCRIPTION
**Pull request by MayXuQQ**
[ipi-azue]: Base Profile - Azure IPI private cluster
<BLANK LINE>
[OCPQE-6472](https://issues.redhat.com/browse/OCPQE-6472)
<BLANK LINE>
<footer>

### CODE REVIEW

**Comment by Amoghrd** [comment]
Is the vnet chain required? If yes, then we have to update all relevant workflows

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/private/deprovision/cucushift-installer-rehearse-azure-ipi-private-deprovision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-private-deprovision
+  steps: 
+  - ref: gather-proxy
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/private/deprovision/cucushift-installer-rehearse-azure-ipi-private-deprovision-chain.yaml:4]: suggest to add  the following here together:
```
- chain: gather-network
- chain: gather-core-dump
```

**Comment by jianlinliu** [comment]
pls always squash multiple commits into one

File: ci-operator/step-registry/ipi/conf/azure/resourcegroup/ipi-conf-azure-resourcegroup-commands.sh [hunk 2]
```diff
@@ -10,7 +10,15 @@ curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -
 CONFIG="${SHARED_DIR}/install-config.yaml"
 PATCH="/tmp/install-config-existingresourcegroup.yaml.patch"
 
-azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
+if [[ ! -z ${AZURE_REGION:-} ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/resourcegroup/ipi-conf-azure-resourcegroup-commands.sh:13]: No, this is not how prow ci work. If you want  a random region selected by prow ci, they are controlled by `azure-qe-quota-slice` defined in `core-services/prow/02_config/_boskos.yaml`

**Pull request merged**